From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 15 Oct 2025 11:27:23 -0600
Subject: [PATCH] Fix MC-54109: Wolfs collar is orange instead of red


diff --git a/src/main/java/net/minecraft/server/EntityWolf.java b/src/main/java/net/minecraft/server/EntityWolf.java
index 8fde8671fbbeb5223d3d40f0b749c3d8008bee85..becafe415d182729d3ab4ccde11e22c7e480f1e4 100644
--- a/src/main/java/net/minecraft/server/EntityWolf.java
+++ b/src/main/java/net/minecraft/server/EntityWolf.java
@@ -89,7 +89,7 @@ public class EntityWolf extends EntityTameableAnimal {
         super.h();
         this.datawatcher.a(18, new Float(this.getHealth()));
         this.datawatcher.a(19, new Byte((byte) 0));
-        this.datawatcher.a(20, new Byte((byte) EnumColor.RED.getColorIndex()));
+        this.datawatcher.a(20, new Byte((byte) (world.dashSpigotConfig.wolfDefaultCollarRed ? EnumColor.RED.getInvColorIndex() : EnumColor.ORANGE.getInvColorIndex()))); // DashSpigot - Fix MC-54109: Wolf's collar is orange instead of red
     }
 
     protected void a(BlockPosition blockposition, Block block) {
diff --git a/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java b/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
index 507a3131cbc8552532a93e3a48baef295d06daae..8468d7afe1b5f2e5f51c08751d080ac7b67ca855 100644
--- a/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
+++ b/src/main/java/xyz/dashnetwork/dashspigot/config/DashSpigotWorldConfig.java
@@ -13,4 +13,7 @@ public class DashSpigotWorldConfig {
     @Comment("Prevent wolves from despawning when tamed. Fixes MC-28424")
     public boolean preventWolvesFromDespawning = true;
 
+    @Comment("Sets the default wolf collar color to be red. Fixes MC-54109")
+    public boolean wolfDefaultCollarRed = true;
+
 }
