From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Sun, 2 Nov 2025 16:44:39 -0700
Subject: [PATCH] Port ArrowCollideEvent


diff --git a/src/main/java/net/minecraft/server/EntityArrow.java b/src/main/java/net/minecraft/server/EntityArrow.java
index b3c1b9c394e1cc4e7e823fdf537b33f07a68d77d..76e8532ca386a18377025d40580e85f35bdf5d5e 100644
--- a/src/main/java/net/minecraft/server/EntityArrow.java
+++ b/src/main/java/net/minecraft/server/EntityArrow.java
@@ -232,7 +232,10 @@ public class EntityArrow extends Entity implements IProjectile {
 
             // PandaSpigot start - Call ProjectileCollideEvent
             if (movingobjectposition != null && movingobjectposition.entity != null) {
-                com.destroystokyo.paper.event.entity.ProjectileCollideEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callProjectileCollideEvent(this, movingobjectposition);
+                // DashSpigot start - Port ArrowCollideEvent
+                net.techcable.tacospigot.event.entity.ArrowCollideEvent event = new net.techcable.tacospigot.event.entity.ArrowCollideEvent((org.bukkit.entity.Arrow) this.getBukkitEntity(), movingobjectposition.entity.getBukkitEntity());
+                org.bukkit.Bukkit.getPluginManager().callEvent(event);
+                // DashSpigot end
                 if (event.isCancelled()) {
                     movingobjectposition = null;
                 }
