From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <github@masterdash5.com>
Date: Wed, 12 Nov 2025 04:08:44 -0700
Subject: [PATCH] Reset airTicks on respawn


diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
index 8bb792b3a5e9fb1ee15f3cf37f96000c4bb0a74b..ba48cefd38f8b5428c97006fba09e589cfd9c040 100644
--- a/src/main/java/net/minecraft/server/EntityPlayer.java
+++ b/src/main/java/net/minecraft/server/EntityPlayer.java
@@ -1223,6 +1223,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
 
         this.setHealth(this.getMaxHealth());
         this.fireTicks = 0;
+        this.setAirTicks(300); // DashSpigot - Reset airTicks on respawn
         this.fallDistance = 0;
         this.foodData = new FoodMetaData(this);
         this.expLevel = this.newLevel;
