From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 18 Feb 2026 07:52:24 -0700
Subject: [PATCH] use valueOf() for DataWatcher


diff --git a/src/main/java/net/minecraft/server/EntityBat.java b/src/main/java/net/minecraft/server/EntityBat.java
index abe696e5519f450de0c2e4e1b10ba776608510c3..462fa9db710f81f1e47df3532fcb6ac7c6e094d6 100644
--- a/src/main/java/net/minecraft/server/EntityBat.java
+++ b/src/main/java/net/minecraft/server/EntityBat.java
@@ -15,7 +15,7 @@ public class EntityBat extends EntityAmbient {
     @Override
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // DashSpigot - Use valueOf()
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/server/EntityBlaze.java b/src/main/java/net/minecraft/server/EntityBlaze.java
index 720b0d227520e56e2c30cf460c01849a4f0c13c1..257d131116c6db425485693d7515ef9b853a9688 100644
--- a/src/main/java/net/minecraft/server/EntityBlaze.java
+++ b/src/main/java/net/minecraft/server/EntityBlaze.java
@@ -29,7 +29,7 @@ public class EntityBlaze extends EntityMonster {
     @Override
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // DashSpigot - use valueOf()
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/server/EntityBoat.java b/src/main/java/net/minecraft/server/EntityBoat.java
index 0e6f8c03501dd6ce5bda003ae54729763201ebd3..516f12b62646ca3bfe341a3184178f5d2a3cb5ee 100644
--- a/src/main/java/net/minecraft/server/EntityBoat.java
+++ b/src/main/java/net/minecraft/server/EntityBoat.java
@@ -57,9 +57,11 @@ public class EntityBoat extends Entity {
     }
 
     protected void h() {
-        this.datawatcher.a(17, new Integer(0));
-        this.datawatcher.a(18, new Integer(1));
-        this.datawatcher.a(19, new Float(0.0F));
+        // DashSpigot start - use valueOf()
+        this.datawatcher.a(17, Integer.valueOf(0));
+        this.datawatcher.a(18, Integer.valueOf(1));
+        this.datawatcher.a(19, Float.valueOf(0.0F));
+        // DashSpigot end
     }
 
     public AxisAlignedBB j(Entity entity) {
diff --git a/src/main/java/net/minecraft/server/EntityEnderman.java b/src/main/java/net/minecraft/server/EntityEnderman.java
index 6d5fec85c8a79df2ea299c31196bab58f722fa9e..f90ea04a7ea55b5a0cd917e49adea8f5c568bce0 100644
--- a/src/main/java/net/minecraft/server/EntityEnderman.java
+++ b/src/main/java/net/minecraft/server/EntityEnderman.java
@@ -55,9 +55,11 @@ public class EntityEnderman extends EntityMonster {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Short((short) 0));
-        this.datawatcher.a(17, new Byte((byte) 0));
-        this.datawatcher.a(18, new Byte((byte) 0));
+        // DashSpigot start - use valueOf()
+        this.datawatcher.a(16, Short.valueOf((short) 0));
+        this.datawatcher.a(17, Byte.valueOf((byte) 0));
+        this.datawatcher.a(18, Byte.valueOf((byte) 0));
+        // DashSpigot end
     }
 
     public void b(NBTTagCompound nbttagcompound) {
diff --git a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
index 4bf790cdffdbc8950450644a813d9c7bb539793d..aada40dccc53ddba3e6fce6425ba68b0c6f6c0fb 100644
--- a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
+++ b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
@@ -72,11 +72,13 @@ public abstract class EntityMinecartAbstract extends Entity implements INamableT
     }
 
     protected void h() {
-        this.datawatcher.a(17, new Integer(0));
-        this.datawatcher.a(18, new Integer(1));
-        this.datawatcher.a(19, new Float(0.0F));
-        this.datawatcher.a(20, new Integer(0));
-        this.datawatcher.a(21, new Integer(6));
+        // DashSpigot start - use valueOf()
+        this.datawatcher.a(17, Integer.valueOf(0));
+        this.datawatcher.a(18, Integer.valueOf(1));
+        this.datawatcher.a(19, Float.valueOf(0.0F));
+        this.datawatcher.a(20, Integer.valueOf(0));
+        this.datawatcher.a(21, Integer.valueOf(6));
+        // DashSpigot end
         this.datawatcher.a(22, Byte.valueOf((byte) 0));
     }
 
diff --git a/src/main/java/net/minecraft/server/EntityMinecartFurnace.java b/src/main/java/net/minecraft/server/EntityMinecartFurnace.java
index 908620d41116f7d0474b11ec823acf8222bebe80..9390c7db6dd3fb60b7a11244c42b927b60e9f397 100644
--- a/src/main/java/net/minecraft/server/EntityMinecartFurnace.java
+++ b/src/main/java/net/minecraft/server/EntityMinecartFurnace.java
@@ -20,7 +20,7 @@ public class EntityMinecartFurnace extends EntityMinecartAbstract {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // DashSpigot - use valueOf()
     }
 
     public void t_() {
diff --git a/src/main/java/net/minecraft/server/EntitySheep.java b/src/main/java/net/minecraft/server/EntitySheep.java
index 29611a194b8a582d57159231091c66adb7c6944f..43e8910cd8dc69796be24ae5e96768aed1ed4bd0 100644
--- a/src/main/java/net/minecraft/server/EntitySheep.java
+++ b/src/main/java/net/minecraft/server/EntitySheep.java
@@ -71,7 +71,7 @@ public class EntitySheep extends EntityAnimal {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // DashSpigot - use valueOf()
     }
 
     protected void dropDeathLoot(boolean flag, int i) {
diff --git a/src/main/java/net/minecraft/server/EntitySkeleton.java b/src/main/java/net/minecraft/server/EntitySkeleton.java
index 46bff95d8bf73e34eb8e0a9b94fb9fa9e9cc5cfb..06ab8b54a33c9282c48f20c9975267957c911a42 100644
--- a/src/main/java/net/minecraft/server/EntitySkeleton.java
+++ b/src/main/java/net/minecraft/server/EntitySkeleton.java
@@ -34,7 +34,7 @@ public class EntitySkeleton extends EntityMonster implements IRangedEntity {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(13, new Byte((byte) 0));
+        this.datawatcher.a(13, Byte.valueOf((byte) 0)); // DashSpigot - use valueOf()
     }
 
     protected String z() {
diff --git a/src/main/java/net/minecraft/server/EntitySpider.java b/src/main/java/net/minecraft/server/EntitySpider.java
index 68a253c9698d2b239e400d7a9406c4430072cc33..76a991275eab501d5359566b284bbe356d2eb1bb 100644
--- a/src/main/java/net/minecraft/server/EntitySpider.java
+++ b/src/main/java/net/minecraft/server/EntitySpider.java
@@ -29,7 +29,7 @@ public class EntitySpider extends EntityMonster {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(16, new Byte((byte) 0));
+        this.datawatcher.a(16, Byte.valueOf((byte) 0)); // DashSpigot - use valueOf()
     }
 
     public void t_() {
diff --git a/src/main/java/net/minecraft/server/EntityWither.java b/src/main/java/net/minecraft/server/EntityWither.java
index fb19bad3c01b03991f9dbaff6041acbccda3c8b3..be47cec16863c5db55753d691a0e879babe92b96 100644
--- a/src/main/java/net/minecraft/server/EntityWither.java
+++ b/src/main/java/net/minecraft/server/EntityWither.java
@@ -48,10 +48,12 @@ public class EntityWither extends EntityMonster implements IRangedEntity {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(17, new Integer(0));
-        this.datawatcher.a(18, new Integer(0));
-        this.datawatcher.a(19, new Integer(0));
-        this.datawatcher.a(20, new Integer(0));
+        // DashSpigot start - use valueOf()
+        this.datawatcher.a(17, Integer.valueOf(0));
+        this.datawatcher.a(18, Integer.valueOf(0));
+        this.datawatcher.a(19, Integer.valueOf(0));
+        this.datawatcher.a(20, Integer.valueOf(0));
+        // DashSpigot end
     }
 
     public void b(NBTTagCompound nbttagcompound) {
diff --git a/src/main/java/net/minecraft/server/EntityWolf.java b/src/main/java/net/minecraft/server/EntityWolf.java
index becafe415d182729d3ab4ccde11e22c7e480f1e4..7e83ead75a939932fd1a10ad9026d615956dc43f 100644
--- a/src/main/java/net/minecraft/server/EntityWolf.java
+++ b/src/main/java/net/minecraft/server/EntityWolf.java
@@ -87,9 +87,11 @@ public class EntityWolf extends EntityTameableAnimal {
 
     protected void h() {
         super.h();
-        this.datawatcher.a(18, new Float(this.getHealth()));
-        this.datawatcher.a(19, new Byte((byte) 0));
-        this.datawatcher.a(20, new Byte((byte) (world.dashSpigotConfig.wolfDefaultCollarRed ? EnumColor.RED.getInvColorIndex() : EnumColor.ORANGE.getInvColorIndex()))); // DashSpigot - Fix MC-54109: Wolf's collar is orange instead of red
+        // DashSpigot start - use valueOf()
+        this.datawatcher.a(18, Float.valueOf(this.getHealth()));
+        this.datawatcher.a(19, Byte.valueOf((byte) 0));
+        this.datawatcher.a(20, Byte.valueOf((byte) (world.dashSpigotConfig.wolfDefaultCollarRed ? EnumColor.RED.getInvColorIndex() : EnumColor.ORANGE.getInvColorIndex()))); // DashSpigot - Fix MC-54109: Wolf's collar is orange instead of red
+        // DashSpigot end
     }
 
     protected void a(BlockPosition blockposition, Block block) {
