From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Tue, 11 Nov 2025 12:18:53 -0700
Subject: [PATCH] Fix book issues


diff --git a/src/main/java/net/minecraft/server/PacketPlayOutSetSlot.java b/src/main/java/net/minecraft/server/PacketPlayOutSetSlot.java
index e86e1f53ca39e5788fac5d7885990dbad1d102fe..e272c6527f18161ec4a870fa6af026ac65835e02 100644
--- a/src/main/java/net/minecraft/server/PacketPlayOutSetSlot.java
+++ b/src/main/java/net/minecraft/server/PacketPlayOutSetSlot.java
@@ -14,6 +14,14 @@ public class PacketPlayOutSetSlot implements Packet<PacketListenerPlayOut> {
         this.a = i;
         this.b = j;
         this.c = itemstack == null ? null : itemstack.cloneItemStack();
+
+        // DashSpigot start - Fix MC-1685: Unable to write in a blank Book and Quill if contains NBT data
+        if (c != null && c.getItem() == Items.WRITABLE_BOOK && c.hasTag() && c.getTag().getList("pages", 8).isEmpty()) {
+            NBTTagList pages = new NBTTagList();
+            pages.add(new NBTTagString(""));
+            c.getTag().set("pages", pages);
+        }
+        // DashSpigot end
     }
 
     public void a(PacketListenerPlayOut packetlistenerplayout) {
diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index b43051f6b61cbfbaa1492cc45171180f6514d50a..ad61e19fa4d7b8bd6fd1b1b73127c3cff716b427 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -2053,7 +2053,7 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                     }
                 }
             }
-            byteAllowed += (maxBookPageSize * Math.min(1, Math.max(0.1D, (double) length / 255D))) * multiplier;
+            byteAllowed += (long) ((maxBookPageSize * Math.min(1, Math.max(0.1D, (double) length / 255D))) * multiplier); // DashSpigot - Explicit long cast
 
             if (multibytes > 1) {
                 // penalize MB
@@ -2102,7 +2102,7 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                 if (itemstack1 != null) {
                     if (itemstack.getItem() == Items.WRITABLE_BOOK && itemstack.getItem() == itemstack1.getItem()) {
                         if (!validateBook(itemstack)) return; // PandaSpigot
-                        itemstack1 = new ItemStack(Items.WRITABLE_BOOK); // CraftBukkit
+                        //itemstack1 = new ItemStack(Items.WRITABLE_BOOK); // CraftBukkit // DashSpigot - Don't remove exiting NBT
                         itemstack1.a("pages", (NBTBase) itemstack.getTag().getList("pages", 8));
                         CraftEventFactory.handleEditBookEvent(player, itemstack1); // CraftBukkit
                     }
@@ -2146,7 +2146,15 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                         itemstack1 = new ItemStack(Items.WRITTEN_BOOK);
                         itemstack1.a("author", (NBTBase) (new NBTTagString(this.player.getName())));
                         itemstack1.a("title", (NBTBase) (new NBTTagString(itemstack.getTag().getString("title"))));
-                        itemstack1.a("pages", (NBTBase) itemstack.getTag().getList("pages", 8));
+                        // DashSpigot start - Improve book security
+                        NBTTagList pages = itemstack.getTag().getList("pages", 8);
+                        for (int i = 0; i < pages.size(); i++) {
+                            pages.a(i, new NBTTagString(IChatBaseComponent.ChatSerializer.a(
+                                new ChatComponentText(pages.getString(i)).setChatModifier(null)
+                            )));
+                        }
+                        itemstack1.a("pages", pages);
+                        // DashSpigot end
                         itemstack1.setItem(Items.WRITTEN_BOOK);
                         CraftEventFactory.handleEditBookEvent(player, itemstack1);
                         // CraftBukkit end
