From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Tue, 11 Nov 2025 12:18:53 -0700
Subject: [PATCH] Fix book renaming issues


diff --git a/src/main/java/net/minecraft/server/PacketPlayOutSetSlot.java b/src/main/java/net/minecraft/server/PacketPlayOutSetSlot.java
index e86e1f53ca39e5788fac5d7885990dbad1d102fe..e272c6527f18161ec4a870fa6af026ac65835e02 100644
--- a/src/main/java/net/minecraft/server/PacketPlayOutSetSlot.java
+++ b/src/main/java/net/minecraft/server/PacketPlayOutSetSlot.java
@@ -14,6 +14,14 @@ public class PacketPlayOutSetSlot implements Packet<PacketListenerPlayOut> {
         this.a = i;
         this.b = j;
         this.c = itemstack == null ? null : itemstack.cloneItemStack();
+
+        // DashSpigot start - Fix MC-1685: Unable to write in a blank Book and Quill if contains NBT data
+        if (c != null && c.getItem() == Items.WRITABLE_BOOK && c.hasTag() && c.getTag().getList("pages", 8).isEmpty()) {
+            NBTTagList pages = new NBTTagList();
+            pages.add(new NBTTagString(""));
+            c.getTag().set("pages", pages);
+        }
+        // DashSpigot end
     }
 
     public void a(PacketListenerPlayOut packetlistenerplayout) {
