From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MasterDash5 <constant4337@molecularmail.com>
Date: Wed, 10 Dec 2025 20:19:42 -0700
Subject: [PATCH] fix it for real this time


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index ad61e19fa4d7b8bd6fd1b1b73127c3cff716b427..a52db6be82711f1d238a2a5172ec8df6499cae85 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -2149,9 +2149,7 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                         // DashSpigot start - Improve book security
                         NBTTagList pages = itemstack.getTag().getList("pages", 8);
                         for (int i = 0; i < pages.size(); i++) {
-                            pages.a(i, new NBTTagString(IChatBaseComponent.ChatSerializer.a(
-                                new ChatComponentText(pages.getString(i)).setChatModifier(null)
-                            )));
+                            pages.a(i, new NBTTagString(IChatBaseComponent.ChatSerializer.a(pages.getString(i)).getText()));
                         }
                         itemstack1.a("pages", pages);
                         // DashSpigot end
